### PR TITLE
Remove repetition in definition of var(u)int types

### PR DESCRIPTION
**Pull request by rossberg-chromium**

### CODE REVIEW

File: BinaryEncoding.md [hunk 1]
```diff
@@ -31,26 +31,21 @@ for a proposal for layer 1 structural compression.
...
+
+### varuintN
+A [LEB128](https://en.wikipedia.org/wiki/LEB128) variable-length integer, limited to _N_ bits (i.e., the values [0, 2^_N_-1]),
+represented by at most trunc(_N_/7)+1 bytes that may contain padding zero bytes.
+Currently, the only sizes _N_ used are 1, 7, and 32,
+where the former are used for compatibility with potential future extensions.
+
+### varintN
+A [Signed LEB128](https://en.wikipedia.org/wiki/LEB128#Signed_LEB128) variable-length integer, limited to _N_ bits (i.e., the values [-2^(_N_-1), +2^(_N_-1)-1]),
+represented by at most trunc(_N_/7)+1 bytes that may contain padding zero or all-one bytes.
```
Comment (titzer) [inline BinaryEncoding.md]: I think we don't allow padding bytes for varuint1 and varuint7.

Comment (rossberg-chromium) [inline BinaryEncoding.md]: That's implied by the at most trunc(N/7)+1 bytes. Except that I just realise that the formula is off-by-one for N=7. :)

Comment (rossberg-chromium) [inline BinaryEncoding.md]: Fixed & simplified.

File: BinaryEncoding.md [hunk 2]
```diff
@@ -31,26 +31,21 @@ for a proposal for layer 1 structural compression.
...
-
-### varuint32
-A [LEB128](https://en.wikipedia.org/wiki/LEB128) variable-length integer, limited to uint32 values. `varuint32` values may contain leading zeros and must be at most 5 bytes.
-
-### varint64
-A [Signed LEB128](https://en.wikipedia.org/wiki/LEB128#Signed_LEB128) variable-length integer, limited to int64 values. `varint64` values may contain leading zero or one bits and must be at most 10 bytes.
+### uintN
+An unsigned integer of _N_ bits,
+represented in _N_/8 bytes in [little endian](https://en.wikipedia.org/wiki/Endianness#Little-endian) order.
+_N_ is either 8, 16, or 32.
```
Comment (binji) [inline BinaryEncoding.md]: Maybe name the currently supported types explicitly here, for better searchability?
i.e. uint8, uint16, uint32.

File: BinaryEncoding.md [hunk 3]
```diff
@@ -31,26 +31,21 @@ for a proposal for layer 1 structural compression.
...
+
+### varuintN
+A [LEB128](https://en.wikipedia.org/wiki/LEB128) variable-length integer, limited to _N_ bits (i.e., the values [0, 2^_N_-1]),
+represented by at most trunc(_N_/7)+1 bytes that may contain padding zero bytes.
+Currently, the only sizes _N_ used are 1, 7, and 32,
+where the former are used for compatibility with potential future extensions.
```
Comment (titzer) [inline BinaryEncoding.md]: s/where the former/where 1 and 7/

File: BinaryEncoding.md (hunk 2 above)
Comment (rossberg-chromium) [inline BinaryEncoding.md]: Done.

**Comment by rossberg-chromium** [comment]
Ping

**Comment by titzer** [comment]
lgtm

**Pull request merged**